polyfill/intersection-observer.js

```diff
@@ -726,7 +726,7 @@ function getParentNode(node) {
 
 
 // Exposes the constructors globally.
-window.IntersectionObserver = IntersectionObserver;
-window.IntersectionObserverEntry = IntersectionObserverEntry;
+root.IntersectionObserver = IntersectionObserver;
+root.IntersectionObserverEntry = IntersectionObserverEntry;
 
-}(window, document));
+}(typeof window !== "undefined" ? window : this));
```

Comment:
In the future, this line could be modified to support `globalThis`:

```suggestion
}((
  typeof globalThis !== "undefined"
    ? globalThis
    : typeof window !== "undefined"
       ? window
        : this
));
```

**Note:** I would prefer a version which does _not_ use nested ternary statements, so this is for illustration purposes only.